engine/src/main/java/com/google/android/fhir/search/NestedSearch.kt

```diff
@@ -61,16 +85,17 @@ internal fun List<NestedSearch>.nestedQuery(
   return if (isEmpty()) {
     null
   } else {
-    map { it.nestedQuery(type) }.let {
-      SearchQuery(
-        query =
-          it.joinToString(
-            prefix = "AND a.resourceUuid IN ",
-            separator = " ${operation.logicalOperator} a.resourceUuid IN"
-          ) { "(\n${it.query}\n) " },
-        args = it.flatMap { it.args }
-      )
-    }
+    map { it.nestedQuery(type) }
+      .let {
+        SearchQuery(
+          query =
+            it.joinToString(
+              prefix = "AND a.resourceUuid IN ",
+              separator = " ${operation.logicalOperator} a.resourceUuid IN"
+            ) { "(\n${it.query}\n) " },
+          args = it.flatMap { it.args }
+        )
+      }
```

Comment:
since we are using `it` multiple times, let us explicitly name the outer one:

```suggestion
      .let { searchQueries ->
        SearchQuery(
          query =
            searchQueries.joinToString(
              prefix = "AND a.resourceUuid IN ",
              separator = " ${operation.logicalOperator} a.resourceUuid IN"
            ) { "(\n${it.query}\n) " },
          args = searchQueries.flatMap { it.args }
        )
      }
```